### PR TITLE
ekf: add `altlog` testing utility

### PR DESCRIPTION
**Pull request by niewim19**
JIRA: PP-69

## Description

`altlog` is ekf/sensorhub utility that allows to test ekf altitude estimation possibilities. It is possible to specify total sampling time and sampling interval to suite test needs. It instantiates `ekf` thread and communicates with sensorhub for direct pressure data.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: zturn drone

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: ekf/tests/altlog.c [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+			if (strcmp("-h", argv[1]) == 0) {
+				altlog_help();
+				return EXIT_SUCCESS;
+			}
+			tmax = (float)atoi(argv[1]);
+			break;
+		default:
+			altlog_help();
+			return EXIT_FAILURE;
+			break;
```
Comment (HBuczynski) [inline ekf/tests/altlog.c:68]: This approach seems to be prone to errors. If You add some new argument You will have to copy again case for `-h`.
In this code, during copying-pasting, You left useless `break;` in `line:68`.
I recommend to use `getopt` for parsing command-line arguments.

File: ekf/tests/altlog.c [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+#include <string.h>
+#include <sys/time.h>
+
+#include <ekflib.h>
+#include <sensc.h>
+
+
+static void altlog_help(void)
+{
+	printf("Usage: altlog dur [tstep]\n");
```
Comment (HBuczynski) [inline ekf/tests/altlog.c:29]: Help usually takes `const char* progname` (`argv[0]`) to avoid harcoding program name

File: ekf/tests/altlog.c [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+
+static float altlog_timeSec(void)
+{
+	time_t now;
+
+	gettime(&now, NULL);
+
+	return (float)now / 1000000;
```
Comment (HBuczynski) [inline ekf/tests/altlog.c:41]:
```suggestion
	return (float)now / 1000000.0;
```

File: ekf/tests/altlog.c [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+		return EXIT_FAILURE;
+	}
+	ekf_run();
+
+	t0 = altlog_timeSec();
+	do {
+		ekf_stateGet(&uavState);
+		sensc_baroGet(&baroEvt);
+
+		gettime(&now, NULL);
```
Comment (HBuczynski) [inline ekf/tests/altlog.c:95]: `now` is not used

**Review by HBuczynski — CHANGES_REQUESTED** [review]

**Comment by niewim19** [comment]
## Not yet ready for review.

Remark about using `getopt` untouched!

**Comment by niewim19** [comment]
Ready for review

File: ekf/tests/altlog.c [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+				break;
+
+			default: /* any other case is an error */
+				altlog_help(argv[0]);
+				ret = -1;
+		}
+	}
+
+	/* make it microseconds for usleep */
+	altlog_common.step *= 1000;
```
Comment (HBuczynski) [inline ekf/tests/altlog.c:91]: Lack of initialization of `altlog_common.step`. If program is invoked without  `-s`, the `altlog_common.step = 0`.

File: ekf/tests/altlog.c [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+				break;
+
+			case 's': /* samppling interval set */
+				altlog_common.step = atoi(optarg);
+				ret = (altlog_common.step == 0) ? -1 : 0;
+				break;
+
+			default: /* any other case is an error */
+				altlog_help(argv[0]);
+				ret = -1;
```
Comment (HBuczynski) [inline ekf/tests/altlog.c:86]: Missing `break;`

File: ekf/tests/altlog.c [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+	return (float)now / 1000000.0;
+}
+
+
+/*
+* Parsing arguments Returns:
+*  0: no error,
+*  1: no error but program should halt,
+* -1: error occurred.
```
Comment (HBuczynski) [inline ekf/tests/altlog.c:56]: Adding variable `altlog_common.init` which is set if mandatory data is parsed will definitely reduce the number of `if` conditions in the code.

**Review by HBuczynski — CHANGES_REQUESTED** [review]

**Review by HBuczynski — APPROVED** [review]

**Pull request merged**